libsweep/sweep_device.hpp

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include "protocol.h"
+#include "serial.h"
+
+#include <chrono>
+#include <string>
+#include <thread>
+
+#include "sweep.h"
+#include "sweep_device.hpp"
+
+#define SWEEP_MAX_SAMPLES 4096
+
+struct sweep_scan {
+  int32_t angle[SWEEP_MAX_SAMPLES];           // in millidegrees
+  int32_t distance[SWEEP_MAX_SAMPLES];        // in cm
+  int32_t signal_strength[SWEEP_MAX_SAMPLES]; // range 0:255
+  int32_t count;
+};
+
+struct sweep_device {
+public:
+  sweep_device();
+  sweep_device(const char* port, int32_t bitrate);
+  ~sweep_device();
+
+  void start_scanning();
+  void stop_scanning();
+
+  sweep_scan_s get_scan();
+
+  int32_t get_motor_speed();
+  void set_motor_speed(int32_t hz);
+
+  void reset();
+
+private:
+  sweep::serial::Device serial; // serial port communication
```

Comment:
Sorry for the confusion: `sweep_device` is not meant to be part part of the public API. The public c++ API is still `Sweep` and the public C-api still consists of the functions in `sweep.h` and the opaque pointer. Admittedly I still think that it is unnecessary to provide a stable ABI for c++ projects, but that is not my decision and